tools/state-viewer/src/state_parts.rs

```diff
@@ -446,31 +447,43 @@ trait StatePartReader {
 fn get_state_part_reader(
```

Comment:
This file can be refactored to remove `StatePartReader` and `StatePartWriter`, but it's fine to do it in a followup PR.